wagtail/admin/widgets.py

```diff
@@ -46,7 +47,10 @@ def __init__(self, attrs=None, format=None):
         if attrs:
             default_attrs.update(attrs)
         if fmt is None:
-            fmt = getattr(settings, 'WAGTAIL_DATE_FORMAT', DEFAULT_DATE_FORMAT)
+            if getattr(settings, 'USE_L10N', False):
+                fmt = get_format('DATE_INPUT_FORMATS')[0]
```

Comment:
Letting `DATE_INPUT_FORMATS` take precedence over `WAGTAIL_DATE_FORMAT` would make `WAGTAIL_DATE_FORMAT`'s behaviour consistent with Django's `DATE_FORMAT`, but it is a breaking change that would affect existing projects (see for example #4846 which involves setting Wagtail's format to an entry which is not the first one in the list). Will need to ponder some more to decide if this is a positive change or not - worth noting that this would rule out the possibility of setting Wagtail's format to something different to the Django default (to allow dates to be presented differently in the Wagtail admin vs the site frontend).

Reply:
I see what you mean, but from my testing at least, using `USE_L10N=True` with a `WAGTAIL_DATE_FORMAT` (and without this override) just made the field unusable as the validation would break - the advice in the issues I've seen is to turn that setting off to fix it, which seems awkward? Happy to be corrected on this - perhaps I've had something setup improperly, but that's why I added this L10N specific override.

Reply:
My understanding is that `WAGTAIL_DATE_FORMAT` does work with `USE_L10N=True` enabled, but [as per the docs](https://docs.wagtail.io/en/stable/reference/settings.html#date-and-datetime-inputs) the chosen format needs to be one of the formats in the `DATE_INPUT_FORMATS` list, and with `USE_L10N` enabled there are extra hoops to jump through to find out what that list is (and doubly so if you actually want to override it).

What we could do is add a startup check that warns if `WAGTAIL_DATE_FORMAT` has been set to something not on the `DATE_INPUT_FORMATS` list, and perhaps have our documentation emphasise a bit more that if you intend to change `WAGTAIL_DATE_FORMAT` from its default, you'll probably have to faff around with locales...?